modules/lwjgl/core/src/main/java/org/lwjgl/system/ThreadLocalUtil.java

```diff
@@ -184,21 +189,17 @@ public static void setCapabilities(long capabilities) {
 
     // Ensures FUNCTION_MISSING_ABORT will be called even if no context is current,
     public static void setFunctionMissingAddresses(int functionCount) {
-        // OpenJDK: NULL
-        // GraalVM Native Image: pointer to UnimplementedWithJNIEnvArgument function (see #875)
-        long RESERVED0_NULL = memGetAddress(JNI_NATIVE_INTERFACE);
-
         long ptr = JNI_NATIVE_INTERFACE + CAPABILITIES_OFFSET;
 
         long currentTable = memGetAddress(ptr);
         if (functionCount == 0) {
-            if (currentTable != RESERVED0_NULL) {
+            if (currentTable != RESERVED3_NULL) {
```

Comment:
What about checking for `currentTable == FUNCTION_MISSING_ABORT_TABLE` (i.e., "is it the value we set").
And resetting it to `RESERVED3_NULL` instead of `NULL` (i.e., reset it to "the value that was there before").